app/src/main/java/com/example/bottomnav/Components.java

```diff
@@ -5,39 +5,107 @@
+        if (dec.getDeclSpecifier() instanceof CPPASTSimpleDeclSpecifier) {
+            return makeOptComp((CPPASTSimpleDeclSpecifier) dec.getDeclSpecifier(),
+                    (CPPASTDeclarator) dec.getDeclarators()[0]);
+        } else {
+            return makeOptComp((CPPASTNamedTypeSpecifier) dec.getDeclSpecifier(),
+                    (CPPASTDeclarator) dec.getDeclarators()[0]);
+        }
+    }
+
+    /**
+     *
+     * @param simpleSpec
+     * @param declarator
+     * @return Optional<Components>
+     * @throws Exception
+     *
+     * Difference: Simple specifiers don't quantifier types and do not have name object
+     */
+    private static Optional<Components> makeOptComp(CPPASTSimpleDeclSpecifier simpleSpec,
+                                                    CPPASTDeclarator declarator) throws Exception {
+        CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+        Optional<CPPASTEqualsInitializer> initOpt = makeOptInit(init);
+
+        if (declarator != null && simpleSpec != null) {
+            CPPASTName TQname = new CPPASTName();
+            TQname.setName(simpleSpec.getSyntax().getCharImage());
+            CPPASTName primName = new CPPASTName();
+            primName.setName(getPrimitiveType(simpleSpec.getType()));
+            return Optional.of(new Components((CPPASTName) declarator.getName(),
+                    TQname,
+                    primName,
+                    initOpt));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * This parser is a little goofy, instead of returning toString version of prim. type, it
+     * returns a number
+     */
+    private static char[] getPrimitiveType(int primType) {
+        switch (primType) {
+            case 0:
+                return "long".toCharArray();
+            case 3:
+                return "int".toCharArray();
+            case 4:
+                return "float".toCharArray();
+            case 5:
+                return "double".toCharArray();
+            default:
+                return null;
+        }
+    }
+
+    /**
+     *
+     * @param namedTypeSpec
+     * @param declarator
+     * @return Optional<Components>
+     * @throws Exception
+     *
+     * Difference: Name typed specifiers have quantifer types and have name object
+     */
+    private static Optional<Components> makeOptComp(CPPASTNamedTypeSpecifier namedTypeSpec,
+                                                    CPPASTDeclarator declarator) throws Exception {
         CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
-        Optional<CPPASTEqualsInitializer> initOpt = makeOpt(init);
-        if (declarator != null && specifier != null && declaration != null) {
+        Optional<CPPASTEqualsInitializer> initOpt = makeOptInit(init);
+        CPPASTName TQname = new CPPASTName();
```

Comment:
This is a bit weird that you have to materialize a new AST node. Can you refactor your interfaces so you don't have to do this? For example, instead of defining Component on AST nodes, define it using the unpacked data like strings?